### DISCUSSION TITLE
enable onError without enabling alpha API

### DISCUSSION THREAD

**Discussion by pritidesai** [tektoncd/pipeline#4239] [Q&A]
Pipeline [0.27](https://github.com/tektoncd/pipeline/releases/tag/v0.27.0) was release in August which introduced an alpha feature to specify `onError` in a `step`. This feature was implemented in the PR #4106. To enable this feature, the Tekton Pipeline deployment has to set `enable-api-fields` to `alpha`.

In our deployment, we have many use cases which require the functionality `onError` provides but cannot enable `alpha` APIs. 

What do folks think about introducing an additional feature flag to just enable `onError` functionality? 

An alternative is to promote it to stable/beta API. Based on our [guidelines](https://github.com/tektoncd/community/blob/main/teps/0033-tekton-feature-gates.md#promoting-a-field-from-alpha-to-the-highest-stable-level), promotion can be done after a few releases(3 to 4). Can we file an exception for this feature?

/cc @skaegi @bobcatfish @vdemeester @afrittoli @sbwsg @jerop

**Comment by jerop** [comment]
I prefer adding an additional feature flag to promoting it to beta faster so that we have room to easily make any necessary changes based on user feedback

More broadly, I wonder if we should have guidelines for these circumstances considering the cons of individual feature flags as discussed in [Feature Gates TEP](https://github.com/tektoncd/community/blob/main/teps/0033-tekton-feature-gates.md#one-feature-flag-per-new-alpha-field), such as:
- do we want to consider adding individual feature flags on a case by case basis?
- do we want to add tooling to show which individual features have been enabled?
- do we want to set expectations for testing coverage of the individual feature with other related features?

**Reply by jerop** [reply]
couldn't find a central documentation for all alpha features so added it here https://github.com/tektoncd/pipeline/pull/4246 - please take a look :)
